galsim/base.py

```diff
@@ -2181,6 +2181,105 @@ def getScaleRadius(self):
         return self.SBProfile.getScaleRadius()
 
 
+class Spergel(GSObject):
+    """A class describing a Spergel profile.
+
+    The Spergel surface brightness profile is characterized by three properties: its Spergel index
+    `nu`, its `flux`, and either the `half_light_radius` or `scale_radius`.  Given these properties,
+    the surface brightness profile scales as I(r) ~ r^{nu} * K_{nu}(r), where K_{nu} is the
+    modified Bessel function of the second kind.
+
+    For more information, refer to
+
+        D. N. Spergel, "ANALYTICAL GALAXY PROFILES FOR PHOTOMETRIC AND LENSING ANALYSIS,"
+        ASTROPHYS J SUPPL S 191(1), 58-65 (2010) [doi:10.1088/0067-0049/191/1/58].
+
+    Initialization
+    --------------
+
+    The allowed range of values for the `nu` parameter is -0.85 <= nu <= 0.85.  An exception will be
+    thrown if you provide a value outside that range.  These values are chosen to match the values
+    investigated by the above paper.  At and below nu=0.0, the surface brightness profile diverges
+    at the origin, which currently affects our ability to perform photon-shooting or `real_space`
+    convolutions.
+
+    The Spergel profile is the Fourier transform of the Moffat profile, and hence has an analytic
+    Fourier transform if not truncated.  If truncation is used, however, then the calculations take
+    advantage of Hankel transform tables that are precomputed for a given value of nu and truncation
+    radius when the Spergel profile is initialized.  Making additional objects with the same nu and
+    truncation radius can therefore be many times faster than making objects with different values
+    of nu and truncation radius that have not been used before.  Moreover, these Hankel transforms
+    are only cached for a maximum of 100 different nu/truncation radius pairs at a time.  For this
+    reason, for large sets of simulations, it is worth considering the use of only discrete
+    nu/truncation radius pairs rather than allowing these to vary continuously.
+
+    A Spergel can be initialized using one (and only one) of two possible size parameters:
+    `scale_radius` or `half_light_radius`.  Exactly one of these two is required.
+
+    @param nu               The Spergel index, nu.
+    @param half_light_radius  The half-light radius of the profile.  Typically given in arcsec.
+                            [One of `scale_radius` or `half_light_radius` is required.]
+    @param scale_radius     The scale radius of the profile.  Typically given in arcsec.
+                            [One of `scale_radius` or `half_light_radius` is required.]
+    @param flux             The flux (in photons) of the profile. [default: 1]
+    @param trunc            An optional truncation radius at which the profile is made to drop to
```

Comment:
Is there a good reason to allow truncation of Spergel profiles?  My understanding was that their main advantage over Sersic is that they have analytic Fourier transforms.  This isn't the case if they are truncated.  So I think maybe we ought not to allow truncation for them.

Unless there is some other advantage of Spergel that would be maintained even if they are truncated.